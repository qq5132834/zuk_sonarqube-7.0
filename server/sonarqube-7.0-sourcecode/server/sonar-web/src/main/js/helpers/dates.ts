/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import {
  differenceInDays as _differenceInDays,
  differenceInSeconds as _differenceInSeconds,
  differenceInYears as _differenceInYears,
  isSameDay as _isSameDay,
  parse,
  startOfDay as _startOfDay
} from 'date-fns';

function pad(number: number) {
  if (number < 10) {
    return '0' + number;
  }
  return number;
}

type ParsableDate = string | number | Date;

export function parseDate(rawDate: ParsableDate): Date {
  return parse(rawDate);
}

export function toShortNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return `${date.getFullYear()}-${pad(date.getMonth() + 1)}-${pad(date.getDate())}`;
}

export function toNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return date.toISOString().replace(/\..+Z$/, '+0000');
}

export function startOfDay(date: Date): Date {
  return _startOfDay(date);
}

export function isValidDate(date: Date): boolean {
  return !isNaN(date.getTime());
}

export function isSameDay(dateLeft: Date, dateRight: Date): boolean {
  return _isSameDay(dateLeft, dateRight);
}

export function differenceInYears(dateLeft: Date, dateRight: Date): number {
  return _differenceInYears(dateLeft, dateRight);
}

export function differenceInDays(dateLeft: Date, dateRight: Date): number {
  return _differenceInDays(dateLeft, dateRight);
}

export function differenceInSeconds(dateLeft: Date, dateRight: Date): number {
  return _differenceInSeconds(dateLeft, dateRight);
}
