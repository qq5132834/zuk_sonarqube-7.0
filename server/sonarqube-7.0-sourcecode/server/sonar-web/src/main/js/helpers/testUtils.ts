/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow, ShallowRendererProps, ShallowWrapper, ReactWrapper } from 'enzyme';
import { IntlProvider } from 'react-intl';

export const mockEvent = {
  target: { blur() {} },
  currentTarget: { blur() {} },
  preventDefault() {},
  stopPropagation() {}
};

export function click(element: ShallowWrapper | ReactWrapper, event = {}): void {
  element.simulate('click', { ...mockEvent, ...event });
}

export function clickOutside(event = {}): void {
  const dispatchedEvent = new MouseEvent('click', event);
  window.dispatchEvent(dispatchedEvent);
}

export function submit(element: ShallowWrapper): void {
  element.simulate('submit', {
    preventDefault() {}
  });
}

export function change(element: ShallowWrapper, value: string, event = {}): void {
  element.simulate('change', {
    target: { value },
    currentTarget: { value },
    ...event
  });
}

export function keydown(keyCode: number): void {
  const event = new KeyboardEvent('keydown', { keyCode } as KeyboardEventInit);
  document.dispatchEvent(event);
}

export function elementKeydown(element: ShallowWrapper, keyCode: number): void {
  const event = {
    currentTarget: { element },
    keyCode,
    preventDefault() {}
  };

  if (typeof element.type() === 'string') {
    // `type()` is string for native dom elements
    element.simulate('keydown', event);
  } else {
    element.prop<Function>('onKeyDown')(event);
  }
}

export function doAsync(fn?: Function): Promise<void> {
  return new Promise(resolve => {
    setImmediate(() => {
      if (fn) {
        fn();
      }
      resolve();
    });
  });
}

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider({ locale: 'en' }, {});
const { intl } = intlProvider.getChildContext();
export function shallowWithIntl(node: React.ReactElement<any>, options: ShallowRendererProps = {}) {
  return shallow(node, { ...options, context: { intl, ...options.context } });
}
