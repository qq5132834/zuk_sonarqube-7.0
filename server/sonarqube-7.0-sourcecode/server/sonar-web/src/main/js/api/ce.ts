/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { getJSON, post, RequestData } from '../helpers/request';
import throwGlobalError from '../app/utils/throwGlobalError';

export interface PendingTask {
  componentKey: string;
  componentName: string;
  componentQualifier: string;
  id: string;
  logs: boolean;
  organization: string;
  status: string;
  submittedAt: Date;
  submitterLogin?: string;
  type: string;
}

export interface Task extends PendingTask {
  analysisId?: string;
  errorMessage?: string;
  errorType?: string;
  executionTimeMs: number;
  executedAt: Date;
  hasErrorStacktrace: boolean;
  hasScannerContext: boolean;
  startedAt: Date;
}

export function getActivity(data: RequestData): Promise<any> {
  return getJSON('/api/ce/activity', data);
}

export function getStatus(componentId?: string): Promise<any> {
  const data = {};
  if (componentId) {
    Object.assign(data, { componentId });
  }
  return getJSON('/api/ce/activity_status', data);
}

export function getTask(id: string, additionalFields?: string[]): Promise<any> {
  return getJSON('/api/ce/task', { id, additionalFields }).then(r => r.task);
}

export function cancelTask(id: string): Promise<any> {
  return post('/api/ce/cancel', { id }).then(() => getTask(id), () => getTask(id));
}

export function cancelAllTasks(): Promise<any> {
  return post('/api/ce/cancel_all');
}

export function getTasksForComponent(
  componentKey: string
): Promise<{ queue: PendingTask[]; current: Task }> {
  return getJSON('/api/ce/component', { componentKey }).catch(throwGlobalError);
}

export function getTypes(): Promise<any> {
  return getJSON('/api/ce/task_types').then(r => r.taskTypes);
}

export function getWorkers(): Promise<{ canSetWorkerCount: boolean; value: number }> {
  return getJSON('/api/ce/worker_count').catch(throwGlobalError);
}

export function setWorkerCount(count: number): Promise<void | Response> {
  return post('/api/ce/set_worker_count', { count }).catch(throwGlobalError);
}
