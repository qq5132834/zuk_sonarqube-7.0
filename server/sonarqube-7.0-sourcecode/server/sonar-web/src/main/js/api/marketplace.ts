/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { checkStatus, corsRequest, getJSON, parseJSON, post, postJSON } from '../helpers/request';
import throwGlobalError from '../app/utils/throwGlobalError';

export interface Edition {
  key: string;
  name: string;
  textDescription: string;
  homeUrl: string;
  licenseRequestUrl: string;
  downloadUrl: string;
}

export interface EditionsPerVersion {
  [version: string]: Edition[];
}

export interface EditionStatus {
  currentEditionKey?: string;
  nextEditionKey?: string;
  installError?: string;
  installationStatus:
    | 'NONE'
    | 'AUTOMATIC_IN_PROGRESS'
    | 'MANUAL_IN_PROGRESS'
    | 'AUTOMATIC_READY'
    | 'UNINSTALL_IN_PROGRESS';
}

export function getEditionStatus(): Promise<EditionStatus> {
  return getJSON('/api/editions/status');
}

export function getEditionsList(url: string): Promise<EditionsPerVersion> {
  return corsRequest(url)
    .submit()
    .then(checkStatus)
    .then(parseJSON);
}

export function getLicensePreview(data: {
  license: string;
}): Promise<{
  nextEditionKey: string;
  previewStatus: 'NO_INSTALL' | 'AUTOMATIC_INSTALL' | 'MANUAL_INSTALL';
}> {
  return postJSON('/api/editions/preview', data).catch(throwGlobalError);
}

export function getFormData(): Promise<{ serverId: string; ncloc: number }> {
  return getJSON('/api/editions/form_data').catch(throwGlobalError);
}

export function applyLicense(data: { license: string }): Promise<EditionStatus> {
  return postJSON('/api/editions/apply_license', data).catch(throwGlobalError);
}

export function uninstallEdition(): Promise<void | Response> {
  return post('/api/editions/uninstall').catch(throwGlobalError);
}

export function dismissErrorMessage(): Promise<void | Response> {
  return post('/api/editions/clear_error_message').catch(throwGlobalError);
}
