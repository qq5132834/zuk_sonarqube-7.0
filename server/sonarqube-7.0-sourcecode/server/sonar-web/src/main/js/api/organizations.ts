/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { getJSON, post, postJSON, RequestData } from '../helpers/request';
import throwGlobalError from '../app/utils/throwGlobalError';

interface GetOrganizationsParameters {
  organizations?: string;
  member?: boolean;
}

interface GetOrganizationsResponse {
  organizations: Array<{
    avatar?: string;
    description?: string;
    guarded: boolean;
    isAdmin: boolean;
    key: string;
    name: string;
    url?: string;
  }>;
  paging: {
    pageIndex: number;
    pageSize: number;
    total: number;
  };
}

export function getOrganizations(
  data: GetOrganizationsParameters
): Promise<GetOrganizationsResponse> {
  return getJSON('/api/organizations/search', data);
}

export function getOrganization(key: string): Promise<any> {
  return getOrganizations({ organizations: key })
    .then(r => r.organizations.find((o: any) => o.key === key))
    .catch(throwGlobalError);
}

interface GetOrganizationNavigation {
  canAdmin: boolean;
  canDelete: boolean;
  canProvisionProjects: boolean;
  isDefault: boolean;
  pages: Array<{ key: string; name: string }>;
  adminPages: Array<{ key: string; name: string }>;
}

export function getOrganizationNavigation(key: string): Promise<GetOrganizationNavigation> {
  return getJSON('/api/navigation/organization', { organization: key }).then(r => r.organization);
}

export function createOrganization(data: RequestData): Promise<any> {
  return postJSON('/api/organizations/create', data).then(r => r.organization, throwGlobalError);
}

export function updateOrganization(key: string, changes: RequestData): Promise<void> {
  return post('/api/organizations/update', { key, ...changes });
}

export function deleteOrganization(key: string): Promise<void | Response> {
  return post('/api/organizations/delete', { key }).catch(throwGlobalError);
}

export function searchMembers(data: {
  organization?: string;
  p?: number;
  ps?: number;
  q?: string;
  selected?: string;
}): Promise<any> {
  return getJSON('/api/organizations/search_members', data);
}

export function addMember(data: { login: string; organization: string }): Promise<any> {
  return postJSON('/api/organizations/add_member', data).then(r => r.user);
}

export function removeMember(data: { login: string; organization: string }): Promise<void> {
  return post('/api/organizations/remove_member', data);
}

export function changeProjectVisibility(
  organization: string,
  projectVisibility: string
): Promise<void> {
  return post('/api/organizations/update_project_visibility', { organization, projectVisibility });
}
