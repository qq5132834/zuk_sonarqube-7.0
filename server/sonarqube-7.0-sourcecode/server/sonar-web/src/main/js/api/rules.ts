/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { post, getJSON, RequestData } from '../helpers/request';
import throwGlobalError from '../app/utils/throwGlobalError';

export interface GetRulesAppResponse {
  respositories: Array<{ key: string; language: string; name: string }>;
}

export function getRulesApp(): Promise<GetRulesAppResponse> {
  return getJSON('/api/rules/app').catch(throwGlobalError);
}

export function searchRules(data: RequestData) {
  return getJSON('/api/rules/search', data).catch(throwGlobalError);
}

export function takeFacet(response: any, property: string) {
  const facet = response.facets.find((facet: any) => facet.property === property);
  return facet ? facet.values : [];
}

export interface GetRuleDetailsParameters {
  actives?: boolean;
  key: string;
  organization?: string;
}

export function getRuleDetails(parameters: GetRuleDetailsParameters): Promise<any> {
  return getJSON('/api/rules/show', parameters).catch(throwGlobalError);
}

export function getRuleTags(parameters: { organization?: string }): Promise<string[]> {
  return getJSON('/api/rules/tags', parameters).then(r => r.tags, throwGlobalError);
}

export function deleteRule(parameters: { key: string }) {
  return post('/api/rules/delete', parameters).catch(throwGlobalError);
}
