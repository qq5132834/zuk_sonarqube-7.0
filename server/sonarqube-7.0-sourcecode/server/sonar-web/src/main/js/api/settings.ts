/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { omitBy } from 'lodash';
import { getJSON, RequestData, post, postJSON } from '../helpers/request';
import { TYPE_PROPERTY_SET } from '../apps/settings/constants';
import throwGlobalError from '../app/utils/throwGlobalError';

export function getDefinitions(component: string | null, branch?: string): Promise<any> {
  return getJSON('/api/settings/list_definitions', { branch, component }).then(r => r.definitions);
}

export interface SettingValue {
  inherited?: boolean;
  key: string;
  parentValue?: string;
  parentValues?: string[];
  value?: any;
  values?: string[];
}

export function getValues(
  keys: string,
  component?: string,
  branch?: string
): Promise<SettingValue[]> {
  return getJSON('/api/settings/values', { keys, component, branch }).then(r => r.settings);
}

export function setSettingValue(
  definition: any,
  value: any,
  component?: string,
  branch?: string
): Promise<void> {
  const { key } = definition;
  const data: RequestData = { key, component, branch };

  if (definition.multiValues) {
    data.values = value;
  } else if (definition.type === TYPE_PROPERTY_SET) {
    data.fieldValues = value
      .map((fields: any) => omitBy(fields, value => value == null))
      .map(JSON.stringify);
  } else {
    data.value = value;
  }

  return post('/api/settings/set', data);
}

export function setSimpleSettingValue(parameters: {
  branch?: string;
  component?: string;
  value: string;
  key: string;
}): Promise<void | Response> {
  return post('/api/settings/set', parameters).catch(throwGlobalError);
}

export function resetSettingValue(key: string, component?: string, branch?: string): Promise<void> {
  return post('/api/settings/reset', { keys: key, component, branch });
}

export function sendTestEmail(to: string, subject: string, message: string): Promise<void> {
  return post('/api/emails/send', { to, subject, message });
}

export function checkSecretKey(): Promise<any> {
  return getJSON('/api/settings/check_secret_key');
}

export function generateSecretKey(): Promise<any> {
  return postJSON('/api/settings/generate_secret_key');
}

export function encryptValue(value: string): Promise<any> {
  return postJSON('/api/settings/encrypt', { value });
}
