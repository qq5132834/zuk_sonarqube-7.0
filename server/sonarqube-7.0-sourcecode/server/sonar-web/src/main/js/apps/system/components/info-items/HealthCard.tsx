/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import * as classNames from 'classnames';
import { map } from 'lodash';
import HealthItem from './HealthItem';
import OpenCloseIcon from '../../../../components/icons-components/OpenCloseIcon';
import Section from './Section';
import { HealthType, SysValueObject } from '../../../../api/system';
import { LOGS_LEVELS, groupSections, getLogsLevel } from '../../utils';
import { translate } from '../../../../helpers/l10n';

interface Props {
  biggerHealth?: boolean;
  health?: HealthType;
  healthCauses?: string[];
  onClick: (toggledCard: string) => void;
  open: boolean;
  name: string;
  sysInfoData: SysValueObject;
}

interface State {
  hoveringDetail: boolean;
}

export default class HealthCard extends React.PureComponent<Props, State> {
  state: State = { hoveringDetail: false };

  handleClick = () => this.props.onClick(this.props.name);
  onDetailEnter = () => this.setState({ hoveringDetail: true });
  onDetailLeave = () => this.setState({ hoveringDetail: false });

  render() {
    const { health, open, sysInfoData } = this.props;
    const { mainSection, sections } = groupSections(sysInfoData);
    const showFields = open && mainSection && Object.keys(mainSection).length > 0;
    const showSections = open && sections;
    const logLevel = getLogsLevel(sysInfoData);
    const showLogLevelWarning = logLevel && logLevel !== LOGS_LEVELS[0];
    return (
      <li
        className={classNames('boxed-group system-info-health-card', {
          'no-hover': this.state.hoveringDetail
        })}>
        <div className="boxed-group-header" onClick={this.handleClick}>
          <span className="system-info-health-card-title">
            <OpenCloseIcon className="little-spacer-right" open={open} />
            {this.props.name}
          </span>
          {showLogLevelWarning && (
            <span className="alert alert-danger spacer-left">
              {translate('system.log_level.warning.short')}
            </span>
          )}
          {health && (
            <HealthItem
              biggerHealth={this.props.biggerHealth}
              className="pull-right"
              health={health}
              healthCauses={this.props.healthCauses}
              name={this.props.name}
            />
          )}
        </div>
        {open && (
          <div
            className="boxed-group-inner"
            onMouseEnter={this.onDetailEnter}
            onMouseLeave={this.onDetailLeave}>
            {showFields && <Section items={mainSection} />}
            {showSections &&
              map(sections, (section, name) => <Section key={name} items={section} name={name} />)}
          </div>
        )}
      </li>
    );
  }
}
