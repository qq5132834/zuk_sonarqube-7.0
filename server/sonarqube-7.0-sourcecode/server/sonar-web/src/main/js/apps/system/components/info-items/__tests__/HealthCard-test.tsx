/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import HealthCard from '../HealthCard';
import { click } from '../../../../../helpers/testUtils';
import { HealthType } from '../../../../../api/system';

it('should render correctly', () => {
  expect(getShallowWrapper()).toMatchSnapshot();
});

it('should display the sysinfo detail', () => {
  expect(getShallowWrapper({ biggerHealth: true, open: true })).toMatchSnapshot();
});

it('should show the sysinfo detail when the card is clicked', () => {
  const onClick = jest.fn();
  click(getShallowWrapper({ onClick }).find('.boxed-group-header'));
  expect(onClick).toBeCalled();
  expect(onClick).toBeCalledWith('Foobar');
});

it('should show a main section and multiple sub sections', () => {
  const sysInfoData = {
    Name: 'foo',
    bar: 'Bar',
    Database: { db: 'test' },
    Elasticseach: { Elastic: 'search' }
  };
  expect(getShallowWrapper({ open: true, sysInfoData })).toMatchSnapshot();
});

it('should display the log level alert', () => {
  expect(
    getShallowWrapper({ sysInfoData: { 'Logs Level': 'DEBUG' } }).find('.alert')
  ).toMatchSnapshot();
});

function getShallowWrapper(props = {}) {
  return shallow(
    <HealthCard
      biggerHealth={false}
      health={HealthType.RED}
      healthCauses={['foo']}
      name="Foobar"
      onClick={() => {}}
      open={false}
      sysInfoData={{}}
      {...props}
    />
  );
}
