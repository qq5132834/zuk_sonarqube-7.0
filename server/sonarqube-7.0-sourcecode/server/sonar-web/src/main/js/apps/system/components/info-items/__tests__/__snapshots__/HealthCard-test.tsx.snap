// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the log level alert 1`] = `
<span
  className="alert alert-danger spacer-left"
>
  system.log_level.warning.short
</span>
`;

exports[`should display the sysinfo detail 1`] = `
<li
  className="boxed-group system-info-health-card"
>
  <div
    className="boxed-group-header"
    onClick={[Function]}
  >
    <span
      className="system-info-health-card-title"
    >
      <OpenCloseIcon
        className="little-spacer-right"
        open={true}
      />
      Foobar
    </span>
    <HealthItem
      biggerHealth={true}
      className="pull-right"
      health="RED"
      healthCauses={
        Array [
          "foo",
        ]
      }
      name="Foobar"
    />
  </div>
  <div
    className="boxed-group-inner"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</li>
`;

exports[`should render correctly 1`] = `
<li
  className="boxed-group system-info-health-card"
>
  <div
    className="boxed-group-header"
    onClick={[Function]}
  >
    <span
      className="system-info-health-card-title"
    >
      <OpenCloseIcon
        className="little-spacer-right"
        open={false}
      />
      Foobar
    </span>
    <HealthItem
      biggerHealth={false}
      className="pull-right"
      health="RED"
      healthCauses={
        Array [
          "foo",
        ]
      }
      name="Foobar"
    />
  </div>
</li>
`;

exports[`should show a main section and multiple sub sections 1`] = `
<li
  className="boxed-group system-info-health-card"
>
  <div
    className="boxed-group-header"
    onClick={[Function]}
  >
    <span
      className="system-info-health-card-title"
    >
      <OpenCloseIcon
        className="little-spacer-right"
        open={true}
      />
      Foobar
    </span>
    <HealthItem
      biggerHealth={false}
      className="pull-right"
      health="RED"
      healthCauses={
        Array [
          "foo",
        ]
      }
      name="Foobar"
    />
  </div>
  <div
    className="boxed-group-inner"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Section
      items={
        Object {
          "Name": "foo",
          "bar": "Bar",
        }
      }
    />
    <Section
      items={
        Object {
          "db": "test",
        }
      }
      key="Database"
      name="Database"
    />
    <Section
      items={
        Object {
          "Elastic": "search",
        }
      }
      key="Elasticseach"
      name="Elasticseach"
    />
  </div>
</li>
`;
